sdks/python/apache_beam/io/gcp/bigtable_io_write.py

```diff
@@ -0,0 +1,146 @@
+"""
+from __future__ import absolute_import
+
+import apache_beam as beam
+from apache_beam.metrics import Metrics
+from apache_beam.transforms.display import DisplayDataItem
+
+try:
+  from google.cloud.bigtable import Client
+  from google.cloud.bigtable.batcher import MutationsBatcher
+except ImportError:
+  pass
+
+
+class WriteToBigtable(beam.DoFn):
+  """ Creates the connector can call and add_row to the batcher using each
+  row in beam pipe line
+
+  :type beam_options: class:`~bigtable_configuration.BigtableConfiguration`
+  :param beam_options: class `~bigtable_configuration.BigtableConfiguration`
+  """
+
+  def __init__(self, beam_options):
+    super(WriteToBigtable, self).__init__(beam_options)
+    self.beam_options = beam_options
+    self.client = None
+    self.instance = None
+    self.table = None
+    self.batcher = None
+    self._app_profile_id = beam_options.app_profile_id
+    self.written = Metrics.counter(self.__class__, 'Written Row')
+
+  def start_bundle(self):
+    if self.client is None:
+      self.client = Client(project=self.beam_options.project_id,
+                           admin=True)
+
+    self.instance = self.client.instance(self.beam_options.instance_id)
+    self.table = self.instance.table(self.beam_options.table_id,
+                                     self.beam_options._app_profile_id)
+    self.batcher = MutationsBatcher(self.table)
+
+  def process(self, row):
+    self.written.inc()
+    self.batcher.mutate(row)
+
+  def finish_bundle(self):
+    result = self.batcher.flush()
+    self.batcher = None
+    return result
+
+  def display_data(self):
+    return {'projectId': DisplayDataItem(self.beam_options.project_id,
+                                         label='Bigtable Project Id'),
+            'instanceId': DisplayDataItem(self.beam_options.instance_id,
+                                          label='Bigtable Instance Id'),
+            'tableId': DisplayDataItem(self.beam_options.table_id,
+                                       label='Bigtable Table Id'),
+            'appProfileId': DisplayDataItem(self._app_profile_id,
+                                            label='Bigtable App Profile Id')
+           }
+
+
+class BigtableConfiguration(object):
+  """ Bigtable configuration variables.
+
+  :type project_id: :class:`str` or :func:`unicode <unicode>`
+  :param project_id: (Optional) The ID of the project which owns the
+    instances, tables and data. If not provided, will
+    attempt to determine from the environment.
+
+  :type instance_id: str
+  :param instance_id: The ID of the instance.
+
+  :type table_id: str
+  :param table_id: The ID of the table.
+  """
+
+  def __init__(self, project_id, instance_id, table_id,
+               app_profile_id=None):
+    self.project_id = project_id
+    self.instance_id = instance_id
+    self.table_id = table_id
+    self.app_profile_id = app_profile_id
+
+
+class BigtableWriteConfiguration(BigtableConfiguration):
```

Comment:
It appears that `BigtableWriteConfiguration` isn't required anymore.  It's basically `BigtableConfiguration` at this point.  Please remove `BigtableWriteConfiguration`